### PR TITLE
Add Invalid QrCode State + Grey Card (EXPOSUREAPP-8373, 8376)

### PR DESCRIPTION
**Pull request by axelherbstreith**
Add invalid state to

- [x] certificate detail screen
- [x] person detail screen
- [x] person overview screen
- [x] Change underlaying qr code when invalid or expired
- [x] Add Grey Card to all certificate detail screens when invalid or expired
- [x] Add Grey state to Person Detail Card

To test: 
- check if the QR code gets greyed + color scheme changes to grey in PersonOverview, PersonDetail and all CertificateDetail Screens if a certificate is invalid or expired

### CODE REVIEW

**Comment by chiljamgossow** [comment]
device-2021-07-29-134257

Found this, not sure whether it's coming from this PR

**Comment by chiljamgossow** [comment]
Should be grey though, as it expired a minute ago (display should de user tz) some on the overview

**Review by chiljamgossow — COMMENTED** [review]
Some issues in the UI ( expiration needs to be in user tz and the element overlaps) but the colors seem to be fine

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt [hunk 1]
```diff
@@ -84,6 +88,13 @@ class PersonDetailsViewModel @AssistedInject constructor(
     private suspend fun assembleList(personCertificates: PersonCertificates, isLoading: Boolean) =
         mutableListOf<CertificateItem>().apply {
             val priorityCertificate = personCertificates.highestPriorityCertificate
+
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:91]: could be shorter when {...}.also { colorShade.postValue(it) }

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/RecoveryCertificateCard.kt [hunk 2]
```diff
@@ -34,11 +34,24 @@ class RecoveryCertificateCard(parent: ViewGroup) :
             certificate.validUntil.toShortDayFormat()
         )
 
+        val bookmarkIcon = if (curItem.certificate.isValid) R.drawable.ic_bookmark_blue else R.drawable.ic_bookmark
         currentCertificate.isVisible = curItem.isCurrentCertificate
+        bookmark.setImageResource(bookmarkIcon)
+
+        val certIcon = when {
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/items/RecoveryCertificateCard.kt:41]: same here ;-)
could be shorter when {...}.also { ... }

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt [hunk 3]
```diff
@@ -55,6 +56,9 @@ class PersonDetailsViewModel @AssistedInject constructor(
 
     val events = SingleLiveEvent<PersonDetailsEvents>()
 
+    private val colorShadeData = MutableLiveData(colorShade)
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:59]: I'd prefer a MutableStateFlow, but that is just personal preference.. (OPTIONAL)

**Review by BMItr — DISMISSED** [review]
nice work 👌  (some optional comments, + the comments from Chilja)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:91]: done

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsViewModel.kt:59]: let us keep it LiveData from me 😃

**Review by BMItr — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**